### ISSUE TITLE
"Insert code" not appearing below code blocks

### ISSUE THREAD

**Issue by ivancea** [microsoft/codetour#150]
The "Insert code" action below a code block isn't appearing in my tours.
I tried it with .ts and .cs files, but it isn't working in any of them.
The tours I tried were created by me; it's my first time trying the extension.

- Platform: Windows 10
- VSCode version: 1.54.3
- CodeTour version: 0.0.46

Screenshot:
![imagen](https://user-images.githubusercontent.com/5280509/111846569-66f46080-8907-11eb-8286-542a31768034.png)

Tour ("test.tour"):
```JSON
{
  "$schema": "https://aka.ms/codetour-schema",
  "title": "Test",
  "steps": [
    {
      "file": "Frontend/src/Config.ts",
      "description": "Test\r\n\r\n```ts\r\nconst a = dev;\r\n```",
      "line": 12
    }
  ],
  "isPrimary": true
}
```

**Comment by lostintangent** [comment]
Hmm, apologies for this issue! Let me take a look at it right now.

**Comment by ivancea** [comment]
Thanks! :)
If you need more info or me to do some test just tell me

**Comment by lostintangent** [comment]
I figured out the issue and will release a fix this evening. Stay tuned!

**Comment by lostintangent** [comment]
This has been fixed in `v0.0.47` 👍🏻  Let me know if you run into any other issues!

**Comment by ivancea** [comment]
Thanks for the fast response and fix, it's working for me now :)
In the other hand, checking the new fenced block regex, I see that fenced block languages with spaces in it may cause a bug, bacause the new regex isn't checking for new lines.
Example of a language with spaces, in a library used by GitHub: https://github.com/github/linguist/blob/master/lib/linguist/languages.yml#L947 (I'm not sure which library uses vscode, but the bug is reproducible now)
Example Tour:
```JSON
{
  "$schema": "https://aka.ms/codetour-schema",
  "title": "Test",
  "steps": [
    {
      "file": "Frontend/src/Config.ts",
      "description": "Test\r\n\r\n```coldfusion html\r\nconst a = dev;\r\n```",
      "line": 13
    }
  ],
  "isPrimary": true
}
```
Rendered as:
![imagen](https://user-images.githubusercontent.com/5280509/111865048-f419d280-8964-11eb-89a8-49ee85009d0a.png)

Generating the code (Note the first line, "html") (Prettied with prettier):
![imagen](https://user-images.githubusercontent.com/5280509/111865056-04ca4880-8965-11eb-849a-9ebb43ca033c.png)


Also, if the closing fenced block line have any non-whitespace characters between the new line and the backticks, it's an invalid fenced block (I think), and it's rendered this way, also showing the language:

Markdown:
```Markdown
```coldfusion html
const a = dev;
error ```
Extra line
```
Rendered as:
![imagen](https://user-images.githubusercontent.com/5280509/111865213-efa1e980-8965-11eb-817e-638bca378048.png)
Note that without that "Extra line", no "Insert code" is rendered, not sure why.

I think it may be fixed with something like:
```TypeScript
const CODE_FENCE_PATTERN = /```[^\n`]*\n([^`]+)\s*\n```/gm;
```
Also changed the language matcher quantifier from a `+` to a `*`, to allow blocks without language, that are also highlighted by VSCode, and I think should work the same way.

**Comment by lostintangent** [comment]
Ah interesting. Yeah I had accounted for languages being multi word. Thanks for the suggestion! I'll get a fix out today 👍

**Comment by lostintangent** [comment]
This is fixed in `v0.0.48`, which is being published to the marketplace right now.

**Comment by ivancea** [comment]
Nice, thanks!
I'm thinking, the `` [^`]+ `` part may be toot strict, and languages that use backticks in code, like JS, won't be able to use them.
I think it isn't feasible with regex, but with a proper parser, so it would be interesting to have that in the baklog, if it isn't already; switching from regex to a Markdown parser.
Everything in `generatePreviewContent` suffers from the same problems.
If you consider it interesting, I could open an issue and work on it (switching from rege to a MD parser there)

**Comment by lostintangent** [comment]
I just switched the regex to use `(.+)` instead, which I think should accommodate many instances. I tested a bunch of different scenarios, and it seems to work pretty well. I agree that at some point a Markdown parser will make the most sense. But in practice, I've noticed that the current regex-based solution seems to be working fairly well for current users. That said, if you're down to submit a PR, then I would love to collaborate on that!
